tests/cpp/test_preseg_passes.cpp

```diff
@@ -642,4 +642,491 @@ TEST_F(PresegTest, ReplaceOutput) {
+  // The partial amax reduction has a single reduction axis.
+  checkAmaxSegmentation(
+      runtime,
+      /*number_of_segments=*/2,
+      /*fusion_index=*/0,
+      /*number_of_outputs_in_fusion=*/3,
+      /*number_of_iterdomains=*/2,
+      /*expected_number_of_reduction_axes=*/1);
+
+  // Aten reference
+  at::Tensor at_x_cast = at_x.to(at::kFloat);
+  at::Tensor at_bias_cast = at_bias.to(at::kFloat);
+  at::Tensor at_residual_cast = at_residual.to(at::kFloat);
+  at::Tensor at_ln_weight_cast = at_ln_weight.to(at::kFloat);
+  at::Tensor at_ln_bias_cast = at_ln_bias.to(at::kFloat);
+  at::Tensor at_t1 = at_x_cast + at_bias_cast.unsqueeze(0);
+  at::Tensor at_t2 = at_t1 + at_residual_cast;
+  at::Tensor at_gamma_centered_ln_weight = at_ln_weight_cast + 1.0f;
+  at::Tensor at_t3 = at::layer_norm(
+      at_t2, {1228}, at_gamma_centered_ln_weight, at_ln_bias_cast, 1e-5);
+  at::Tensor at_t4 = at_t3 * at_fp8_scale;
+  at::Tensor at_t5 = at_t4.to(at::kBFloat16);
+  at::Tensor at_t6 = at::abs(at_t3);
+  at::Tensor at_t7 = at::max(at_t6);
+
+  testValidate(
+      runtime.fusionSegments()->completeFusion(),
+      outputs,
+      aten_inputs,
+      {at_t2, at_t5, at_t7},
+      __LINE__,
+      __FILE__);
+}
+
+TEST_F(NVFuserTest, FusionFactorAmaxHorizontalMultiplePartial_CUDA) {
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(fusion_ptr.get());
+
+  TensorView* tv0 = makeContigTensor(2, DataType::Float);
+  TensorView* tv1 = makeContigTensor(0, DataType::Float);
+  fusion.addInput(tv0);
+  fusion.addInput(tv1);
+
+  // Add Inner and Outer Partial Reductions together
+  TensorView* tv2 = sum(tv0, {1}, /*keepdim=*/true);
+  TensorView* tv3 = sum(tv0, {0}, /*keepdim=*/false);
+  TensorView* tv4 = add(tv2, tv3);
+  fusion.addOutput(tv4);
+
+  // Full Amax Reduction
+  TensorView* tv5 = abs(tv4);
+  TensorView* tv6 = max(tv5, {0, 1}, /*keepdim=*/false);
+
+  // Amax Aliased Output
+  fusion.aliasOutputToInput(tv6, tv1, AllocationType::ReuseBuffer);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor x = at::randn({32, 1228}, options);
+  at::Tensor fp8_amax_history = at::zeros({}, options);
+  std::vector<c10::IValue> aten_inputs = {x, fp8_amax_history};
+
+  auto args = KernelArgumentHolder::createKernelArgumentHolder(aten_inputs);
+  FusionKernelRuntime runtime(std::move(fusion_ptr), args);
+  runtime.compileFusionParallel(args);
+  auto outputs = runtime.runWithInputs(args);
+
+  // Expected result of factorAmaxReduction pass:
+  // * Four segments are created because a partial reduction and a full
+  // reduction cannot be in the same fusion.
+  // * The first two reductions cannot be fused together because they do not
+  // share a common reduction axis.
+  // * Neither of the first two reduction segments can be fused with the third
+  // segment because the reduction scheduler does not support the pointwise
+  // epilogue.
```

Comment:
I think here by `pointwise epilogue.`, you mean broadcast to another TensorView in the epilogue?

Ideally we should fix the scheduler, but adding the logic into the optimization pass might be an easier target.

Reply:
```
**Segmenter** Considering fusion:
T2_l[ iS2{i0}, rS3{i2} ]
   = reduction( T0_g[ iS0{i0}, iS1{i2} ], op = add, initial value = float(0), allreduce = false )
T3_g[ iS4{i0}, bS5{1} ]
   = broadcast( T2_l[ iS2{i0}, rS3{i2} ] )
T6_g[ iS10{i0}, iS11{i2} ]
   = T3_g[ iS4{i0}, bS5{1} ]
   + T5_g[ bS8{1}, iS9{i2} ];

Scheduler _reduction_ ***rejected*** because : has unsupported post reduction fusion
Scheduler _inner_persistent_ ***rejected*** because : no persistent buffer identified
```
This ^^^ is the fusion pattern I was referring to. 

1. Both sum reductions cannot be merged because they have different reduction axes, so two reduction fusions are created.
2. The pointwise addition cannot be fused to either of the reduction fusions because of unsupported epilogue fusion.